apps/judicial-system/web/src/utils/hooks/useIndictmentCounts/index.ts

```diff
@@ -0,0 +1,120 @@
+import { useCallback } from 'react'
+import { useIntl } from 'react-intl'
+import { useMutation } from '@apollo/client'
+
+import { toast } from '@island.is/island-ui/core'
+import { errors } from '@island.is/judicial-system-web/messages'
+import { IndictmentCount } from '@island.is/judicial-system-web/src/graphql/schema'
+
+import { UpdateIndictmentCountMutation } from './updateIndictmentCountGql'
+import { CreateIndictmentCountMutation } from './createIndictmentCountGql'
+import { DeleteIndictmentCountMutation } from './deleteIndictmentCountGql'
+
+interface CreateIndictmentCountMutationResponse {
+  createIndictmentCount: IndictmentCount
+}
+
+interface UpdateIndictmentCountMutationResponse {
+  updateIndictmentCount: {
+    id: string
+  }
+}
+
+interface DeleteIndictmentCountMutationResponse {
+  deleteIndictmentCount: {
+    deleted: boolean
+  }
+}
+
+const useIndictmentCounts = () => {
+  const { formatMessage } = useIntl()
+
+  const [
+    createIndictmentCountMutation,
+  ] = useMutation<CreateIndictmentCountMutationResponse>(
+    CreateIndictmentCountMutation,
+  )
+  const [
+    updateIndictmentCountMutation,
+  ] = useMutation<UpdateIndictmentCountMutationResponse>(
+    UpdateIndictmentCountMutation,
+  )
+
+  const [
+    deleteIndictmentCountMutation,
+  ] = useMutation<DeleteIndictmentCountMutationResponse>(
+    DeleteIndictmentCountMutation,
+  )
+
+  const createIndictmentCount = useCallback(
+    async (caseId: string) => {
+      try {
+        const { data } = await createIndictmentCountMutation({
+          variables: {
+            input: {
+              caseId,
+            },
+          },
+        })
+
+        if (!data) {
+          toast.error(formatMessage(errors.createIndictmentCount))
+        }
+
+        return data?.createIndictmentCount
+      } catch (e) {
+        toast.error(formatMessage(errors.createIndictmentCount))
+      }
+    },
+    [createIndictmentCountMutation, formatMessage],
+  )
+
+  const deleteIndictmentCount = useCallback(
+    async (caseId: string, indictmentCountId: string) => {
+      try {
+        const { data } = await deleteIndictmentCountMutation({
+          variables: {
+            input: {
+              caseId,
+              indictmentCountId,
+            },
+          },
+        })
+
+        return data?.deleteIndictmentCount.deleted
+      } catch (e) {
+        toast.error(formatMessage(errors.deleteIndictmentCount))
+      }
+    },
+    [deleteIndictmentCountMutation, formatMessage],
+  )
+
+  const updateIndictmentCount = useCallback(
+    async (caseId: string, indictmentCount: IndictmentCount) => {
```

Comment:
Consider limiting the type of the second argument to only include fields that can be updated. You could use `pick` or something similar to accomplish this. If you export that type, you can then use the type to limit the allowed fields in the `update` methods in the indictment and indictment count components.